test/test_date_utils.js

```diff
@@ -111,6 +111,29 @@ export default function main() {
     eleven_thirty.setSeconds(0);
     eleven_thirty.setMilliseconds(0);
     test(new DatePiece(null, null, 25, new Time.Absolute(11, 30, 0)), eleven_thirty);
+
+    const next_current_weekday = new Date;
+    const current_weekday_number = next_current_weekday.getDay();
+    const current_weekday_string = next_current_weekday.toLocaleString('en-US', { weekday: 'long' }).toLowerCase();
+    next_current_weekday.setDate(next_current_weekday.getDate() + 1);
+    while (next_current_weekday.getDay() !== current_weekday_number)
+        next_current_weekday.setDate(next_current_weekday.getDate() + 1);
+    next_current_weekday.setHours(0, 0, 0, 0);
+    test(new WeekDayDate(current_weekday_string, new Time.Absolute(0, 0, 0)), next_current_weekday);
+
+    const next_sunday_noon = new Date;
+    next_sunday_noon.setDate(next_sunday_noon.getDate() + 1);
+    while (next_sunday_noon.getDay() !== 0)
+        next_sunday_noon.setDate(next_sunday_noon.getDate() + 1);
+    next_sunday_noon.setHours(12, 0, 0, 0);
+    test(new WeekDayDate('sunday', new Time.Absolute(12, 0, 0)), next_sunday_noon);
+
+    const next_monday_noon = new Date;
+    next_monday_noon.setDate(next_monday_noon.getDate() + 1);
+    while (next_monday_noon.getDay() !== 1)
+        next_monday_noon.setDate(next_monday_noon.getDate() + 1);
+    next_monday_noon.setHours(12, 0, 0, 0);
+    test(new WeekDayDate('monday', new Time.Absolute(12, 0, 0)), next_monday_noon);
```

Comment:
Honestly, I would just loop over all possible days in a month and check all of them...

Reply:
I suppose loop on all date * all weekday?